features/steps/thens/alignment.py

```diff
@@ -352,3 +359,67 @@ def step_impl(context, inst, activation_phrase):
                         observed=observed_msg,
                         severity=OutcomeSeverity.ERROR,
                     )
+
+
+@gherkin_ifc.step('Each segment must have geometric continuity in {continuity_type}')
+def step_impl(context, inst, continuity_type):
+    """
+    Assess geometric continuity between alignment segments for ALS016, ALS017, and ALS018
+    """
+    # Ref: https://standards.buildingsmart.org/IFC/RELEASE/IFC4_3/HTML/lexical/IfcTransitionCode.htm
+    position_transition_codes = ["CONTINUOUS", "CONTSAMEGRADIENT", "CONTSAMEGRADIENTSAMECURVATURE"]
+    tangency_transition_codes = ["CONTSAMEGRADIENT", "CONTSAMEGRADIENTSAMECURVATURE"]
+
+    length_unit_scale_factor = ifcopenshell.util.unit.calculate_unit_scale(
+        ifc_file=context.model,
+        unit_type="LENGTHUNIT"
+    )
+    for pair in inst:
+        pair_continuity_was_calculated = False
+        previous, current = pair
+        entity_contexts = ifc.recurrently_get_entity_attr(context, current, 'IfcRepresentation', 'ContextOfItems')
+        precision = ifc.get_precision_from_contexts(entity_contexts)
+        precision_str = str(precision)
+        parts = precision_str.lower().split("e")
+        mantissa = parts[0]
+        sig_figs = len(mantissa)
+        if "." in mantissa:
+            sig_figs -= 1
+
+        exponent_str = parts[-1]
+        display_sig_figs = int(abs(float(exponent_str)) + sig_figs + 1)  # add one more digit for rounding confirmation
```

Comment:
Would this also work?

```suggestion
      from math import log10, ceil
      display_sig_figs = int(ceil(log10(precision))) + 1
```

```
>>> -log10(0.01)
2.0
```

Reply:
Yes. I have revised accordingly.